media/js/base/mozilla-client.js

```diff
@@ -320,6 +320,71 @@ if (typeof Mozilla === 'undefined') {
         }));
     };
 
+    /**
+     * Use the async mozUITour API of Firefox to retrieve the user's Sync/FxA info. This API
+     * is available on Firefox 35 and later. See
+     * http://bedrock.readthedocs.org/en/latest/uitour.html for details.
+     *
+     * @param  {Function} callback - callback function to be executed with the Sync details
+     * @return {None}
+     */
+    Client.getSyncDetails = function (callback) {
+        // Fire the callback function immediately if cache exists
+        if (Client.SyncDetails) {
+            callback(Client.SyncDetails);
+
+            return;
+        }
+
+        var callbackID = Math.random().toString(36).replace(/[^a-z]+/g, '');
+
+        var listener = function (event) {
+            if (!event.detail || !event.detail.data || event.detail.callbackID !== callbackID) {
+                return;
+            }
+
+            window.clearTimeout(timer);
+            onRetrieved(event.detail.data.setup, event.detail.data.desktopDevices, event.detail.data.mobileDevices, event.detail.data.totalDevices);
+        };
+
+        var onRetrieved = function (setup, desktopDevices, mobileDevices, totalDevices) {
+            document.removeEventListener('mozUITourResponse', listener, false);
+
+            var details = Client.SyncDetails = {
+                'setup': setup,
+                'desktopDevices': desktopDevices,
+                'mobileDevices': mobileDevices,
+                'totalDevices': totalDevices
+            };
+
+            callback(details);
+        };
+
+        // Prepare fallback function in case the API doesn't work
+        var fallback = function () { onRetrieved(false, 0, 0, 0); };
+
+        // If Firefox is old or for Android, call the fallback function immediately because the API is not implemented
+        var userVer = Client._getFirefoxVersion();
+        if (parseFloat(userVer) < 35 || Client._isFirefoxAndroid()) {
```

Comment:
Iirc, querying `sync` has been supported since Australis, which was Firefox 29. It was only `appinfo` that came later in 35.

Reply:
Ah, sweet. Will update.